### ISSUE TITLE
Extraounsous backslash character at end of PYPI artifact

### ISSUE THREAD

**Issue by jasinner** [anchore/syft#1644]
**What happened**:

When scanning this [Python project](https://github.com/thoth-station/micropipenv/archive/refs/tags/v1.0.2.zip) I get the artifact `openapi-spec-validator@0.2.8 \`. Notice the trailing backslash

**What you expected to happen**:

There is no trailing backslash

**Steps to reproduce the issue**:

```
$ syft packages dir:micropipenv-1.0.2/ | grep openapi-spec-validator
 ✔ Indexed micropipenv-1.0.2 
 ✔ Cataloged packages      [478 packages]

openapi-spec-validator     0.2.8                       python  
openapi-spec-validator     0.2.8 \                     python
```

**Anything else we need to know?**:
```

(venv) [jshepher@vpn2-54-150 micropipenv-1.0.2]$ grep '0.2.8 \\' . -R
./tests/data/requirements/extras_marker/requirements_no_indexes.txt:openapi-spec-validator==0.2.8 \
./tests/data/requirements/extras_marker/requirements_no_dev.txt:openapi-spec-validator==0.2.8 \
./tests/data/requirements/extras_marker/requirements_no_comments.txt:openapi-spec-validator==0.2.8 \
./tests/data/requirements/extras/requirements_no_indexes.txt:openapi-spec-validator==0.2.8 \
./tests/data/requirements/extras/requirements_no_dev.txt:openapi-spec-validator==0.2.8 \
./tests/data/requirements/extras/requirements_no_comments.txt:openapi-spec-validator==0.2.8 \
```

These look like test files which are generated by pip-tools

**Environment**:
- Output of `syft version`: syft 0.60.1
- OS (e.g: `cat /etc/os-release` or similar): Fedora release 37 (Thirty Seven)

**Comment by tgerla** [comment]
Hi @jasinner, can you try upgrading Syft to the latest version and trying again? We've made some fixes in this area recently. If it's still a problem let us know.

**Comment by tgerla** [comment]
We've tested this with 0.60 to reproduce and can confirm that this is fixed in Syft 0.74.1.

**Comment by jasinner** [comment]
Thanks @tgerla we're upgraded now and not seeing the issue any longer.
